dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/resourcetable/table/PeriodResourceTable.java

```diff
@@ -101,7 +102,16 @@ public String getCreateTempTableStatement()
             if ( period != null && period.isValid() )
             {
                 final String isoDate = period.getIsoDate();
-                final int year = PeriodType.getCalendar().fromIso( period.getStartDate() ).getYear();
+                int year;
+
+                if ( period.getPeriodType().equalsName( WeeklyPeriodType.NAME ) )
```

Comment:
@larshelge I'm not sure if this is correct, because it only takes in consideration the "normal" weekly type. But what about the other 4 weekly types that we have? Mainly, I'm not sure how the logic should be for these weekly periods.
Let's take `2019-12-31`. The week of this date belongs to 2020, because the week starts on Monday 30 and there are 4 days of the week in 2020.
What should happen if the periodtype is `WeeklySaturdayPeriodType`. My understanding is that the week now starts on Saturday Dec 28 2019, and therefore the weekly period should belong to 2019, since 4 days of the week are in 2019: 28, 29, 30, 31.